### PR TITLE
add upgrade guides

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

### CODE REVIEW

**Comment by benjaminhuo** [comment]
That's looks pretty good 👍
Thanks @huanggze

**Review by benjaminhuo — APPROVED** [review]

File: content/en/docs/upgrade/upgrade-faq.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+linkTitle: "FAQ"
+weight: 150
+---
+
+## How to upgrade Qingcloud CSI after upgrading?
+
+Curently Qingcloud CSI will not be upgraded by KubeKey. You can run the following command to upgrade CSI manually after finishing KubeSphere upgrade:
+
+```
+git clone https://github.com/yunify/qingcloud-csi.git
```
Comment (FeynmanZhou) [inline content/en/docs/upgrade/upgrade-faq.md:15]: How to verify whether the QingCloud CSI upgrade is successful?

Comment (benjaminhuo) [reply content/en/docs/upgrade/upgrade-faq.md:15]: @Ma-Dan any comments here?

File: content/en/docs/upgrade/upgrade-guides.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+- Make sure you read the release notes carefully
+
+{{< notice warning >}}
+In v3.0.0, KubeSphere refactors many of its components such as Fluent Bit Operator, IAM, etc. Make sure you back up any important components in case you heavily customized them but not from console.
+{{</ notice >}}
+
+- Make your upgrade plan. The two upgrading scenarios are documented below.
+
+## Upgrade KubeSpere Only
+The following command brings your KubeSphere to v3.0.0 without touching Kubenetes control planes:
```
[the remark this answers is not in the document]
Comment (benjaminhuo) [reply content/en/docs/upgrade/upgrade-guides.md:30]: Use kk to upgrade kubesphere is ok, it's consistent with other scenarios and will not introduce more confusing steps.

File: content/en/docs/upgrade/upgrade-guides.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+{{< notice note >}} 
+Please refer to the Cluster section of [config-example.yaml](https://github.com/kubesphere/kubekey/blob/master/docs/config-example.md) for more information. 
+{{</ notice >}}
+
+#### Step 3. Upgrade your cluster
+The following command upgrades your cluster to KubeSphere v3.0.0 and Kubernetes v1.17.9 (default):
+
+```
+./kk upgrade --with-kubesphere --with-kubernetes --file config-sample.yaml
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-guides.md:86]: Here I just don't wanna mess up long flags with short flags

File: content/en/docs/upgrade/upgrade-overview.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+linkTitle: "Overview"
+weight: 50
+---
+
+## Kubernetes
+
+KubeSphere v3.0.0 is compatible with Kubernetes 1.15.x, 1.16.x, 1.17.x and 1.18.x:
+
+- If your KubeSphere v2.1.x is installed on Kubernetes 1.15.x+ , you can choose to only upgrade KubeSphere to v3.0.0 or upgrade Kubernetes (to a higher version) and  KubeSphere (to v3.0.0) at the same time.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-overview.md:14]: It is already mentioned in step-by-step upgrading tutorial

File: content/en/docs/upgrade/upgrade-faq.md (hunk 1 above)
Comment (Ma-Dan) [reply content/en/docs/upgrade/upgrade-faq.md:15]: @min-zh Is there any field indicate this upgrade is successful? I watch the controller and daemoset of the csi.

File: content/en/docs/upgrade/upgrade-guides.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+The following command upgrades your single-node cluster to KubeSphere v3.0.0 and Kubernetes v1.17.9 (default):
+
+```
+./kk upgrade --with-kubesphere --with-kubernetes
+```
+
+To upgrade Kubernetes to a specific version, please explicitly provide the version after the flag `--with-kubernetes`. Available versions are:
+
+- v1.15.12
+- v1.16. 8, v1.16.10, v1.16.12, v1.16.13
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-guides.md]: done

File: content/en/docs/upgrade/upgrade-guides.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-guides.md]: done

File: content/en/docs/upgrade/upgrade-faq.md (hunk 1 above)
Comment (Ma-Dan) [reply content/en/docs/upgrade/upgrade-faq.md:15]: Run follwing command to watch csi controller and daemonset is running
```
root@master1:~# kubectl get po -n kube-system | grep csi
csi-qingcloud-controller-56979d46cb-qk9ck   5/5     Running            0          24h
csi-qingcloud-node-4s8n5                    2/2     Running            0          24h
csi-qingcloud-node-65dqn                    2/2     Running            0          24h
csi-qingcloud-node-khk49                    2/2     Running            0          24h
csi-qingcloud-node-nz9q9                    2/2     Running            0          24h
csi-qingcloud-node-pxr56                    2/2     Running            0          24h
csi-qingcloud-node-whqhk                    2/2     Running            0          24h
```

Then run this command to check csi image version is 1.2.x
```
root@master1:~# kubectl get po -n kube-system csi-qingcloud-controller-56979d46cb-qk9ck -ojson | jq '.spec.containers[].image' | grep qingcloud
"csiplugin/csi-qingcloud:v1.2.0-rc.4"
```

**Comment by pixiake** [comment]
A hint could be added: When upgrading with kubernetes, helm will be upgraded from helm2 to helm3. If you want to continue using helm2, please backup it.
`cp /usr/local/bin/helm  /usr/local/bin/helm2`

File: content/en/docs/upgrade/upgrade-faq.md (hunk 1 above)
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-faq.md:15]: updated

File: content/en/docs/upgrade/upgrade-guides.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pixiake) [reply content/en/docs/upgrade/upgrade-guides.md:34]: non-kubekey or non-ansible cluster may not be supported, such as qke.

[the remark this answers is not in the document]
Comment (benjaminhuo) [reply content/en/docs/upgrade/upgrade-guides.md:34]: ok @huanggze would you help to update this doc per feng's suggustion?

[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-guides.md:34]: ok

[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/upgrade/upgrade-guides.md]: updated

File: content/en/docs/upgrade/upgrade-with-kubekey.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+- Make sure you read the release notes carefully
+
+{{< notice warning >}}
+In v3.0.0, KubeSphere refactors many of its components such as Fluent Bit Operator, IAM, etc. Make sure you back up any important components in case you heavily customized them but not from console.
+{{</ notice >}}
+
+- Make your upgrade plan. The two upgrading scenarios are documented below.
+
+## Upgrade KubeSpere Only
```
Comment (FeynmanZhou) [inline content/en/docs/upgrade/upgrade-with-kubekey.md:34]: Typo:
KubeSphere
Kubernetes
image

File: content/en/docs/upgrade/upgrade-with-kubekey.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
+---
+title: "Upgrade with KubeKey"
+keywords: "kubernetes, upgrade, kubesphere, v3.0.0"
+description: "Upgrade KubeSphere with kubekey"
+
+linkTitle: "Upgrade with KubeKey"
+weight: 100
+---
+KubeKey is recommended for users whose Kubernetes was deployed together with KubeSphere.
```
Comment (FeynmanZhou) [inline content/en/docs/upgrade/upgrade-with-kubekey.md:9]: KubeKey is recommended for users whose KubeSphere and Kubernetes were both deployed by KubeSphere Installer. If your Kubernetes cluster was provisioned by yourself, please refer to [Upgrade with ks-installer](xxx).

**Comment by benjaminhuo** [comment]
> A hint could be added: When upgrading with kubernetes, helm will be upgraded from helm2 to helm3. If you want to continue using helm2, please backup it.
> `cp /usr/local/bin/helm /usr/local/bin/helm2`

Good point，@huanggze we could add this tip somewhere?

**Comment by huanggze** [comment]
> Good point，@huanggze we could add this tip somewhere?

image

**Pull request merged**